server/index.js

```diff
@@ -0,0 +1,205 @@
+    headers: headers,
+    agentOptions: {
+      rejectUnauthorized: false
+    },
+    body: {
+      fields: {
+        username: `${username}`,
+        password: `${password}`
+      }
+    }
+  };
+  request(options, async function(error, resp, body) {
+    if (error) {
+      res.send(error);
+    }
+    if (body.records.length > 0) {
+      const payload = { success: false, message: 'user exists' };
+      res.status(422).send(payload);
+      res.send(payload);
+    } else {
+      if (JSON.parse(process.env.HASH_PASSWORD) === true) {
+        bcrypt.hash(
+          req.body.password.trim(),
+          parseInt(process.env.SALT_ROUNDS, 10),
+          (err, hash) => {
+            optionsCreate.body.fields.password = hash;
+            request(optionsCreate, function(error, resp, body) {
+              if (error) {
+                res.send(error);
+              }
+              if (body) {
+                delete body.fields.password;
+                const token = JWT.createToken(body);
+                const payload = { success: true, token: token, user: body };
+                res.status(200).json(payload);
+              } else {
+                const payload = { success: false };
+                res.status(422).send(payload);
+              }
+            });
+          }
+        );
+      } else {
+        request(optionsCreate, function(error, resp, body) {
+          if (error) {
+            res.send(error);
+          }
+          if (body) {
+            const token = JWT.createToken(body);
+            const payload = { success: true, token: token, user: body };
+            res.status(200).json(payload);
+          } else {
+            const payload = { success: false };
+            res.status(422).send(payload);
+          }
+        });
+      }
+    }
+  });
+});
+
+//TODO: Option to Hash
+app.post('/login', (req, res) => {
+  const username = req.body.username;
+  const password = req.body.password;
+
+  const url = `${process.env.AIRTABLE_ENDPOINT_URL}/v0/${process.env.AIRTABLE_BASE_ID}/${process.env.AIRTABLE_USER_TABLE}?filterByFormula=AND(username%3D%22${username}%22%2Cpassword%3D%22${password}%22)`;
+  const urlHash = `${process.env.AIRTABLE_ENDPOINT_URL}/v0/${process.env.AIRTABLE_BASE_ID}/${process.env.AIRTABLE_USER_TABLE}?filterByFormula=username%3D%22${username}%22`;
+  const headers = {
+    authorization: 'Bearer ' + process.env.AIRTABLE_API_KEY,
+    'x-api-version': process.env.AIRTABLE_API_VERSION,
+    'x-airtable-application-id': process.env.AIRTABLE_BASE_ID,
+    'User-Agent': process.env.AIRTABLE_USER_AGENT
+  };
+  var options = {
+    method: 'GET',
+    url: url,
+    json: true,
+    timeout: 5000,
+    headers: headers,
+    agentOptions: {
+      rejectUnauthorized: false
+    }
+  };
+
+  if (JSON.parse(process.env.HASH_PASSWORD) === true) {
+    options.url = urlHash;
+  }
+
+  request(options, function(error, resp, body) {
+    if (error) {
+      res.send(error);
+    }
```

Comment:
Let's use array destructuring here to replace `body.records[0]` in various places:

```
const [user] = body.records;
```

Reply:
done~